storage/gcloud/aio/storage/storage.py

```diff
@@ -78,32 +71,50 @@ def __init__(self, project, service_file, token=None, session=None):
         resp.raise_for_status()
         return await resp.json()
 
-    async def upload(self, bucket, object_name, file_data, headers=None,
-                     session=None):
-        # pylint: disable=too-many-arguments
-        # https://cloud.google.com/storage/docs/json_api/v1/how-tos/simple-upload
+    async def upload(self, bucket: str, object_name: str,
+                     file_data, content_type: str,
+                     headers=None, session: aiohttp.ClientSession = None,
+                     timeout: int = 120, upload_type_resumable_always: bool = False):
         token = await self.token.get()
         url = f'{STORAGE_UPLOAD_API_ROOT}/{bucket}/o'
+
+        if not self.session:
+            self.session = aiohttp.ClientSession(conn_timeout=10,
+                                                 read_timeout=10)
+        session = session or self.session
+
+        data = self.preprocess_data_(file_data)
+        content_length = len(data)
+
         headers = headers or {}
+        headers.update({
+            'Authorization': f'Bearer {token}',
+            'Content-Length': str(content_length),
+            'Content-Type': content_type
+        })
 
+        if len(file_data) > MAX_CONTENT_LENGTH_SIMPLE_UPLOAD or upload_type_resumable_always:
+            response = await self._upload_resumable(url, bucket, object_name, file_data,
+                                                    headers=headers, session=session, timeout=timeout)
+        else:
+            response = await self._upload_simple(url, bucket, object_name, file_data,
+                                                 content_type, headers=headers, session=session,
+                                                 timeout=timeout)
+
+        return response
```

Comment:
No need for the intermediate `response` value, we can just return directly where need be.